src/build_queue.rs

```diff
@@ -348,116 +377,138 @@ impl AsyncBuildQueue {
-                            release.name, release.version
-                        )
-                    }) {
-                    Ok(()) => {
-                        debug!(
-                            "{}-{} added into build queue",
-                            release.name, release.version
-                        );
-                        self.queue_metrics.queued_builds.add(1, &[]);
-                        crates_added += 1;
-                    }
-                    Err(err) => report_error(&err),
-                }
-            }
+    /// Processes crate changes, whether they got yanked or unyanked.
+    async fn process_version_yank_status(
+        &self,
+        conn: &mut AsyncPoolClient,
+        release: &CrateVersion,
+    ) -> Result<()> {
+        // FIXME: delay yanks of crates that have not yet finished building
+        // https://github.com/rust-lang/docs.rs/issues/1934
+        if let Ok(release_version) = Version::parse(&release.version) {
+            self.set_yanked_inner(
+                conn,
+                release.name.as_str(),
+                &release_version,
+                release.yanked,
+            )
+            .await?;
+        }
 
-            let yanked = change.yanked();
-            let unyanked = change.unyanked();
-            if let Some(release) = yanked.or(unyanked) {
-                // FIXME: delay yanks of crates that have not yet finished building
-                // https://github.com/rust-lang/docs.rs/issues/1934
-                if let Ok(release_version) = Version::parse(&release.version)
-                    && let Err(err) = self
-                        .set_yanked_inner(
-                            &mut conn,
-                            release.name.as_str(),
-                            &release_version,
-                            yanked.is_some(),
-                        )
-                        .await
-                {
-                    report_error(&err);
-                }
+        self.queue_crate_invalidation(&release.name).await;
+        Ok(())
+    }
 
-                self.queue_crate_invalidation(&release.name).await;
-            }
-        }
+    async fn process_version_added(
+        &self,
+        conn: &mut AsyncPoolClient,
+        release: &CrateVersion,
+        registry: Option<&str>,
+    ) -> Result<()> {
+        let priority = get_crate_priority(conn, &release.name).await?;
+        let version = &release
+            .version
+            .parse()
+            .context("couldn't parse release version as semver")?;
+        self.add_crate(&release.name, version, priority, registry)
+            .await
+            .with_context(|| {
+                format!(
+                    "failed adding {}-{} into build queue",
+                    release.name, release.version
+                )
+            })?;
+        debug!(
+            "{}-{} added into build queue",
+            release.name, release.version
+        );
```

Comment:
This log was already existing in the previous implementation with string interpolation - even though this is tracing::debug!, not log::debug! based on what I learned in my previous PR - is there any reason as for why we use string interpolation here, or can I change it like this (as well as in `process_version_deleted)?`
```suggestion
        debug!(
                name=%release.name,
                version=%release.version,
               "added into build queue"
            );
```

Reply:
legacy code, feel free to update it